### ISSUE TITLE
Export monitor data (ping, downtimes, etc)

### ISSUE THREAD

**Issue by ternium1** [louislam/uptime-kuma#1093]
### ⚠️ Please verify that this feature request has NOT been suggested before.

- [X] I checked and didn't find similar feature request

### 🏷️ Feature Request Type

Other

### 🔖 Feature description

Export monitor data as JSON, text file, or excel (csv) (to plot in excel?).

### ✔️ Solution

In the dashboard, A button to export the data that opens a modal that lets you export the data as you want (file type, etc.) like shown in the picture (sorry my UI is in French)
![image](https://user-images.githubusercontent.com/32877621/147368871-401582e7-a124-477d-9ba4-c55fef6b0d25.png)
![image](https://user-images.githubusercontent.com/32877621/147368916-2902e13c-54fa-46ac-bbf3-6676cf80ebb0.png)
(with a space between the button, it would be better)

**Comment by rezzorix** [comment]
Very much love to have this.

With a 20second interval opening the 1week monitor chart gets the browser / VM pretty much down to its knee 😆

**Comment by xeruf** [comment]
Yes, and it could even be integrated with something like https://vega.github.io/vega-lite to produce custom charts - in some cases an offline report is expected

**Comment by xeruf** [comment]
I guess exporting directly from the kuma.db residing at `/var/lib/docker/volumes/uptime-kuma/_data/kuma.db` should not be too hard. Filter the `heartbeat` table according to the monitor_id from the `monitor` table and then convert it somehow...

**Comment by xeruf** [comment]
I am currently in the process of writing a script to visualize the data of the db into a strip plot: https://vega.github.io/vega-lite/examples/tick_strip.html
I will push it to a public repo once it's a little more polished :)

**Comment by clavimarmotel** [comment]
Any news on this feature ? It is the exact holy grail of missing parts to turn this into the perfect solution :) .

**Comment by sulbig** [comment]
Upvote from me on this feature request! The historical data could also be used as evidence of an issue (e.g., to an ISP). Having a way to easily export/download event history would be greatly appreciated.

**Comment by CommanderStorm** [comment]
@sulbig please use the 👍🏻 button in the future to not spam people

**Comment by rondamstra** [comment]
Hi folks — I’m exploring a small external add-on for this exact use case: scheduled PDF/CSV reports for Uptime Kuma, with Telegram/webhook delivery and optional client branding.

The goal is not to replace Uptime Kuma or ask maintainers to add more work. I’m trying to see whether an external report generator would be useful for people who currently take screenshots or manually export data.

Sample mock report: https://uptime-kuma-reportbot.mor-online.nl/sample-report
Landing/demo: https://uptime-kuma-reportbot.mor-online.nl/

Questions:

1. Would you use this for yourself, or mainly for client reporting?
2. Is PDF required, or would CSV/HTML report links be enough?
3. Would Telegram/webhook/Slack delivery be useful?
4. Would you prefer hosted SaaS, self-hosted, or both?

If this is unwanted/noisy here, I’m happy to remove the comment.
